# TASK1

In [11]:
temps = [2, 5, 4, 6, 7, 3, 1]
average = sum(temps) / len(temps)
mx = max(temps)
mn = min(temps)
amplitude = maxi - mini
print(f'{average}, {mx}, {mn}, {amplitude}')

4.0, 7, 1, 1100


| Середнє значення | Максимум | Мінімум | Амплітуда |
|------------------|----------|---------|-----------|
|4.0               |7         |1        |6          |

# TASK2

In [10]:
spending_money = [1200, 1500, 800, 600, 1000, 400, 900]
total = sum(spending_money)
average = total / len(spending_money)
mx = max(spending_money)
mn = min(spending_money)
big_spendings = [idx for idx, val in enumerate(spending_money) if val >= 1000]

print(f'{total}, {average}, {mx}, {mn}, {[d+1 for d in big_spendings]}')

6400, 914.2857142857143, 1500, 400, [1, 2, 5]


| Сума | Середнє значення | Максимум | Мінімум | Дні >= 1000 |
|------|------------------|----------|---------|-------------|
|6400  |914,29            |1500      |400      |1, 2, 5      |

# TASK3

In [5]:
from collections import Counter

In [6]:
votes = ["apple", "banana", "apple", "orange", "banana", "apple", "apple", "apple", "orange"]
counts = Counter(votes)
print(f'{counts}, {counts.most_common(3)}')

Counter({'apple': 5, 'banana': 2, 'orange': 2}), [('apple', 5), ('banana', 2), ('orange', 2)]


| apple | banana | orange |
|-------|--------|--------|
|5      |2       |2       |

# TASK4

In [12]:
subjects = [("maths", [80, 92, 75, 60, 88]), ("physics", [70, 65, 90, 85, 78]), ("english", [95, 100, 85, 90, 87])]
for name, scores in subjects:
    avg = sum(scores) / len(scores)
    mx = max(scores)
    mn = min(scores)
    
    print(f"{name}: avg={avg}, max={mx}, min={mn}")

best_avg = max(subjects, key=lambda x: sum(x[1]) / len(x[1]))
print(best_avg)

maths: avg=79.0, max=92, min=60
physics: avg=77.6, max=90, min=65
english: avg=91.4, max=100, min=85
('english', [95, 100, 85, 90, 87])


| Предмет | Середній бал | Максимум | Мінімум |
|---------|--------------|----------|---------|
|maths    |79.0          |92        |60       |
|physics  |77.6          |90        |65       |
|english  |91.4          |100       |85       |